### DISCUSSION TITLE
Modal + external dimmer - suggestions?

### DISCUSSION THREAD

**Discussion by cheesyMan** [fomantic/Fomantic-UI#2020] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Hi, looking for some suggestion about it.

I've a modal that loads data from remote server. The loading process can take up to 2s so I'd like to show a loader to inform users that loading's in progress.

However, as I use a dimmer+loader set, there's an **overlapping** with the _native modal dimmer_.

I've tried to make the external dimmer(+loader) to disappear inside **modal** **onShow() callback**. But this cause a very fast moment when the external dimmer disappears and the modal dimmer is still invisible; the visual effect is like a camera flash, very ugly,

I've tried to make the external dimmer(+loader) to disappear inside **modal onVisible() callback**. In this case, for a a few moments the two dimmers **overlap**, causing the background to be much more darker.
Better than before, but still not ideal.

I'd like to have a smooth transition between the two dimmers so that I cannot visually detect the change.

Any suggestions? Thanks.

--
Alberto

**Comment by lubber-de** [comment, marked as answer]
Use the modals dimmer itself (instead of a separate dimmer, you can customize it via `dimmerSettings`) and only show that via behavior `show dimmer` (instead of `show`). Whenever your loading is done, then show the same modal via behavior `show`. I believe this is what you were looking at
See https://jsfiddle.net/lubber/w6qajtyo/10/

![delayedmodal](https://user-images.githubusercontent.com/18379884/126317977-691f86c8-3417-4886-bf96-f2ff108fbd87.gif)

**Reply by cheesyMan** [reply]
Perfect, I always learn something new about Fomantic-UI api...

Thanks a lot :)
